src/main/java/io/github/hildi/can/service/StandardJavaSerializationService.java

```diff
@@ -0,0 +1,76 @@
+package io.github.hildi.can.service;
+
+import io.github.hildi.can.exceptions.*;
+import io.github.hildi.can.model.User;
+
+import java.io.*;
+
+/**
+ * Created by Serhii Hildi on 15.03.19.
+ */
+public class StandardJavaSerializationService implements SerializationService {
+
+    @Override
+    public void serialize(User user, File file) {
+
+        assertFileExist(file);
+        assertFileWritable(file);
+        assertParamNotNull(user, file);
+
+        try (ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(file))) {
+            out.writeObject(user);
+        } catch (FileNotFoundException e) {
+            throw new SerializationException("Failed to serialize data, reason: file " + file.getName() + " is not found. ", e);
+        } catch (IOException e) {
+            throw new SerializationException("Failed to serialize data. " + file.getName(), e);
+        }
+    }
+
+    @Override
+    public User deserialize(File file) {
+
+        assertFileReadable(file);
+        assertFileNotNull(file);
```

Comment:
in case if the file argument is null, you will get `NullPointerException` inside `assertFileReadable` method.

same for `serialize` method.

seems that all these `not-null` assertions are not covered by unit tests.